### PR TITLE
Fix examples

### PR DESCRIPTION
**Pull request by nside**
examples didn't build out of the box
also `cd examples ; cmake .` didn't work on my system. guessing it was linking from system versions

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -14,14 +14,22 @@ include_directories(../vendor/cxxopts/include)
...
+include_directories(../vendor/capnproto/c++/src)
 link_directories(../vendor/s2geometry)
 link_directories(/usr/local/lib)
 link_directories(../vendor/CRoaring)
+link_directories(../vendor/capnproto)
+link_directories(../vendor/lmdb/libraries/liblmdb/)
+link_directories(../vendor/capnproto/c++/src/capnp/)
+link_directories(../vendor/capnproto/c++/src/kj/)
+link_directories(../vendor/CRoaring/src/)
+link_directories(${OPENSSL_ROOT_DIR}lib/)
```
Comment (bdon) [inline examples/CMakeLists.txt:27]: @nside OPENSSL_ROOT_DIR is something you are setting elsewhere to point to your homebrew/macports installed OpenSSL?

**Comment by nside** [comment]
followed same instructions as your developer guide, passing it via `cmake -DOPENSSL_ROOT_DIR=/opt/homebrew/opt/openssl@1.1/ .`

**Comment by bdon** [comment]
@nside I am actually thinking maybe we should get rid of the CMakeLists in the examples directory and add it as another target to the root CMakeLists, like the test suite. This would make building the examples faster/easier and reduce duplication of the build config, what do you think?

**Comment by bdon** [comment]
Hmm, on the other hand, having an example CMakeLists for users who want to work out of the main tree is also useful. Let me think about this some more

**Comment by nside** [comment]
I think examples that can be built independently/externally are more valuable

**Comment by bdon** [comment]
Thank you!

**Pull request merged**